### PR TITLE
Add TestGetClusterIPDNSandIP and TestGetNodePortDNSandIP unit test

### PR DESCRIPTION
**Pull request by luhaopei**
format code

### CODE REVIEW

File: pkg/yurttunnel/server/serveraddr/addr_test.go [hunk 1]
```diff
@@ -50,3 +56,185 @@ func TestGetDefaultDomainsForSvc(t *testing.T) {
...
+		ips      []net.IP
+	}
+
+	tests := []struct {
+		desc        string
+		svc         corev1.Service
+		expectValue ExpectValue
+	}{
+		{
+			desc: "there is constants.YurttunnelServerExternalAddrKey ips is  192.168.1.2",
```
Comment (charleszheng44) [inline pkg/yurttunnel/server/serveraddr/addr_test.go:191]: "constants.YurttunnelServerExternalAddrKey exists and is associated to IPs ["192.168.1.2"]"

File: pkg/yurttunnel/server/serveraddr/addr_test.go [hunk 2]
```diff
@@ -50,3 +56,185 @@ func TestGetDefaultDomainsForSvc(t *testing.T) {
...
+			expectValue: ExpectValue{
+				ErrorMsg: "there is constants.YurttunnelServerExternalAddrKey ips is  192.168.1.2",
+				ips: []net.IP{
+					net.ParseIP("192.168.1.2"),
+				},
+				dnsNames: []string{},
+			},
+		},
+		{
+			desc: "there is constants.YurttunnelServerExternalAddrKey dnsnames is  www.bing.com:80",
```
Comment (charleszheng44) [inline pkg/yurttunnel/server/serveraddr/addr_test.go:208]: "constants.YurttunnelServerExternalAddrKey exists and is associated to dnsnames ["www.bing.com:80"]"

File: pkg/yurttunnel/server/serveraddr/addr_test.go [hunk 3]
```diff
@@ -50,3 +56,185 @@ func TestGetDefaultDomainsForSvc(t *testing.T) {
...
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			dnsNames, getIPS, err := getClusterIPDNSandIP(&tt.svc)
+			if err != nil {
+				t.Errorf("error is %v\n", err)
+			} else {
+				if !reflect.DeepEqual(dnsNames, tt.expectValue.dnsNames) ||
+					!reflect.DeepEqual(getIPS, tt.expectValue.ips) {
+					t.Error(tt.expectValue.ErrorMsg)
```
Comment (charleszheng44) [inline pkg/yurttunnel/server/serveraddr/addr_test.go:234]: Change it to `t.Error("expect: %v, but got %v")`?

File: pkg/yurttunnel/server/serveraddr/addr_test.go [hunk 4]
```diff
@@ -50,3 +56,185 @@ func TestGetDefaultDomainsForSvc(t *testing.T) {
...
+
+func TestGetClusterIPDNSandIP(t *testing.T) {
+	type ExpectValue struct {
+		ErrorMsg string
+		dnsNames []string
+		ips      []net.IP
+	}
+
+	tests := []struct {
+		desc        string
```
Comment (charleszheng44) [inline pkg/yurttunnel/server/serveraddr/addr_test.go:186]: can we add a test case that contains both dns and IP?

**Review by charleszheng44 — CHANGES_REQUESTED** [review]
The PR looks good except for some minor issues.

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 3 above)
Comment (BearLu) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:234]: OK

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 4 above)
Comment (BearLu) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:186]: This is a good idea, I will add it later

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 2 above)
Comment (BearLu) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:208]: I will modify it later

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 1 above)
Comment (BearLu) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:191]: I will modify it later

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 4 above)
Comment (BearLu) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:186]: I am try it,But I can't  both dns and IP

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 1 above)
Comment (luhaopei) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:191]: I did it

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 2 above)
Comment (luhaopei) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:237]: I did it

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 3 above)
Comment (luhaopei) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:263]: I did it

File: pkg/yurttunnel/server/serveraddr/addr_test.go (hunk 4 above)
Comment (luhaopei) [reply pkg/yurttunnel/server/serveraddr/addr_test.go:186]: I did it

**Review by charleszheng44 — APPROVED** [review]

**Pull request merged**